lib/cosmos/gui/text/ruby_editor.rb

```diff
@@ -76,67 +76,63 @@ def self.create_format(color = Cosmos::BLACK, style='')
 
       RULES = []
       RUBY_KEYWORDS.each do |w|
-        RULES << ['\b'+w+'\b', 0, STYLES['ruby_keyword']]
+        RULES << ['\b'+w+'\b', STYLES['ruby_keyword']]
       end
       Script.private_instance_methods.each do |w|
-        RULES << ['\b'+w.to_s+'\b', 0, STYLES['cosmos_keyword']]
+        RULES << ['\b'+w.to_s+'\b', STYLES['cosmos_keyword']]
       end
       BRACES.each do |b|
-        RULES << ["#{b}", 0, STYLES['brace']]
+        RULES << ["#{b}", STYLES['brace']]
       end
       RULES.concat([
           # 'def' followed by an identifier
-          ['\bdef\b\s*(\w+)', 1, STYLES['method']],
+          ['\bdef\b\s*(\w+)', STYLES['method']],
           # 'class' followed by an identifier
-          ['\bclass\b\s*(\w+)', 1, STYLES['class']],
+          ['\bclass\b\s*(\w+)', STYLES['class']],
           # Ruby symbol
-          [':\b\w+', 0, STYLES['symbol']],
+          [':\b\w+', STYLES['symbol']],
           # Ruby namespace operator
-          ['\b\w+(::\b\w+)+', 0, STYLES['class']],
+          ['\b\w+(::\b\w+)+', STYLES['class']],
           # Ruby global
-          ['\\$\b\w+', 0, STYLES['string']],
+          ['\\$\b\w+', STYLES['string']],
           # Regex, possibly containing escape sequences
-          ["/.*/", 0, STYLES['string']],
+          ["/.*/", STYLES['string']],
           # Double-quoted string, possibly containing escape sequences
-          ['"([^"\\\\]|\\\\.)*"', 0, STYLES['string']],
+          ['"(?:[^"\\\\]|\\\\.)*"', STYLES['string']],
           # Match interpolated strings "blah #{code} blah"
-          ['(\#\\{)[^\\}]*\\}', 1, STYLES['brace']],
-          ['\#\\{[^\\}]*(\\})', 1, STYLES['brace']],
-          ['\#\\{([^\\}]*)\\}', 1, STYLES['normal']],
+          # This comes before single-quote and back-tick because
+          # those strings should not interpolate variables
+          ['(\#\\{)[^\\}]*\\}', STYLES['brace']],
+          ['\#\\{[^\\}]*(\\})', STYLES['brace']],
+          ['\#\\{([^\\}]*)\\}', STYLES['normal']],
           # Single-quoted string, possibly containing escape sequences
-          ["'[^'\\\\]*(\\.[^'\\\\]*)*'", 0, STYLES['string']],
+          ["'[^'\\\\]*(?:\\.[^'\\\\]*)*'", STYLES['string']],
           # Back-tick string, possibly containing escape sequences
-          ["`[^`\\\\]*(\\.[^`\\\\]*)*`", 0, STYLES['string']],
-          # A single # possibly followed by matched quotes
-          ['#(?=([^\'"]*(\'|")[^\'"]*(\'|"))*[^\'"]*$).*', 0, STYLES['comment']],
+          ["`[^`\\\\]*(\\.[^`\\\\]*)*`", STYLES['string']],
+          # Double quoted string | single quoted string | comment
+          # We don't want the first two but we want the last
+          ["\"[^\"]*\"|\'[^\']*\'|(#.*)", STYLES['comment']],
```

Comment:
I read a whole blog post on this regexp syntax. Still not totally sure why it works but it's an awesome way to write without resorting to look aheads and look behinds.